apps/ejabberd/src/mod_commands.erl

```diff
@@ -278,10 +291,30 @@ add_contact(Caller, JabberID, Name, Groups) ->
     CJid = jid:from_binary(Caller),
     mod_roster:set_roster_entry(CJid, JabberID, Name, Groups).
 
+delete_contacts(Caller, ToDelete) ->
+    maybe_delete_contacts(Caller, ToDelete, []).
+
+maybe_delete_contacts(_, [], NotDeleted) -> NotDeleted;
+maybe_delete_contacts(Caller, [H | T], NotDeleted) ->
+    case {jid_exists(Caller, H), delete_contact(Caller, H)} of
+        {_, error} ->
+            maybe_delete_contacts(Caller, T, NotDeleted ++ [H]);
+        {false, _} ->
+            maybe_delete_contacts(Caller, T, NotDeleted ++ [H]);
+        {true, ok} ->
+            maybe_delete_contacts(Caller, T, NotDeleted)
+    end.
+
 delete_contact(Caller, JabberID) ->
     CJid = jid:from_binary(Caller),
     mod_roster:remove_from_roster(CJid, JabberID).
 
+-spec jid_exists(binary(), binary()) -> boolean().
+jid_exists(CJid, Jid) ->
```

Comment:
Are you sure delete_contact will return error when there is no such contact in callers roster? From what I've observed it will not.

Reply:
And that's why I made this check

Reply:
We can also want not to return users that were not deleted because of their 'non-existence' but it's not up to me. If it was I would say it is better to return these ones to client so this check is necessary.

Reply:
What I could do to make it more elegant is to move this check from case clause in 'maybe_delete_contacts' to 'delete_contact', how about that?

Reply:
Yes, you are right. Sorry. :) No need to move this check elsewhere in my opinion.